benchmarks/ALIA/others/array-copysome.sl

```diff
@@ -0,0 +1,40 @@
+(set-logic ALIA)
+
+(synth-inv inv_fun ((a1 (Array Int Int)) (a2 (Array Int Int)) (n Int) (i Int) (z Int)))
+
+(declare-primed-var a1 (Array Int Int))
+(declare-primed-var a2 (Array Int Int))
+(declare-primed-var i Int)
+(declare-primed-var n Int)
+(declare-primed-var z Int)
+
+
+(define-fun pre_fun ((a1 (Array Int Int)) (a2 (Array Int Int)) (n Int) (i Int) (z Int)) Bool
+(and
+    (= i 0)
+    (and (> n 0)
+    (and (< i n)
+    (and (>= z 0) (< z n))))
+))
+
+(define-fun trans_fun ((a1 (Array Int Int)) (a2 (Array Int Int)) (n Int) (i Int) (z Int) (a1! (Array Int Int)) (a2! (Array Int Int)) (n! Int) (i! Int) (z! Int)) Bool
+(and
+    (< i n)
+    (= i! (+ i 1))
+    (and (= n! n)
+    (and (= z! z)
+    (and (= a2! a2)
+    (or (and (= i z)(= (select a1! i)(select a1 i))) (and (not(= i z)) (= a1! (store a1 i (select a2 i))))))))
+))
+
+(define-fun post_fun ((a1 (Array Int Int)) (a2 (Array Int Int)) (n Int) (i Int) (z Int)) Bool
+(or
+   (< i n)
+   (forall ((x Int))
+          (or (not (not (= x z))) 
+              (= (select a1 x) (select a2 x))))
```

Comment:
Hmm, doesn't seem right.

You only iterate from i = 0 to n and copy every index but z, but here you are checking for _all_ x. You need to restrict x to 0 to n as well.

Reply:
would this work?
```
(forall ((x Int)) (=> (and (> x 0) (< x n))
                         (or (not (not (= x z))) (= (select a1 x) (select a2 x)))))
```

Reply:
I think it should be `(> x 0)`. Other than that, looks fine 👍

Reply:
I meant >=